scancode_api/src/main/java/com/ichack/scancode/model/corners/PointInteger.java

```diff
@@ -0,0 +1,22 @@
+package com.ichack.scancode.model.corners;
+
+/**
+ * Extends the generic class Point providing a distanceTo method for double coords
+ */
+public class PointInteger extends Point<Integer> {
```

Comment:
We need to replace the generic class Point, PointInteger and PointDouble with a single class with Number attributes:

I think a quick mock of the class would be:
```java
public class Point {
  private final Number x;
  private final Number y;

  public Point(Number x, Number y) {
    this.x = x;
    this.y = y;
  }

  // Maybe even constructors with double and int (experiment a bit with them and make it look good)
  public Point(double x, double y) {
    // ...
  }

  public Number getDistance(Point other) {
    // use other.x.doubleValue() when calculating
  }

  // Getters here (class is immutable)
  // ...
}
```